### DISCUSSION TITLE
Order By IS NULL

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#5902] [Q&A]
Is there a way to use findAndCount with order by IS NULL?
I have this findAndCount.
```
const [promos, total] = await this.ormRepository.findAndCount(
      {
        tipo: {
          $nin: [EnumTipo.AFINACAO, EnumTipo.CHAMADA],
        },
        deleted_at: null,
        ...(search && {
          $or: [
            { retranca: { $like: `%${search}%` } },
            { codmaterial: { $like: `%${search}%` } },
          ],
        }),
      },
        orderBy: { created_at: 'desc' },
        limit: 20,
        offset: 20 * (page - 1),
      },
    );
    return [promos, Math.ceil(total / 20)];
  }
```

But I'd like to change its orderby to the following partial query: 
` ORDER BY number_field IS NULL desc, bollean_field is TRUE DESC , updated_at DESC`

Must I use query builder?

Thank you!

**Comment by B4nan** [comment]
Try this:

```ts
orderBy: { [raw('number_field IS NULL desc, bollean_field is TRUE DESC , updated_at')]: 'desc' },
```

**Reply by kennethsato** [reply]
I swear we are working on nodejs's upgrade, but it's still WIP
Would raw be available somewhere on v5?

**Reply by B4nan** [reply]
in v5 raw queries were detected automatically, so maybe just try the same with `expr`. no idea if it works in v5 to be honest.

```ts
orderBy: { [expr('number_field IS NULL desc, bollean_field is TRUE DESC , updated_at')]: 'desc' },
```

**Reply by kennethsato** [reply]
Thanks again, @B4nan !!

I tried it as you suggested:
` orderBy: {expr('codmaterial IS NULL DESC, previa IS TRUE DESC , updated_at')]:'desc'}`

But it did not work.
It expected a "big field" labeled as `c0`.`codmaterial IS NULL DESC, previa IS TRUE DESC , updated_at`

Then I tried:
```
 orderBy: {
          [expr('codmaterial IS NULL')]: 'desc',
          [expr('previa IS TRUE')]: 'desc',
          updated_at: 'desc',
        },
```

But it still expect "codmaterial IS NULL" and "previa IS TRUE" to be a real field.

If I could just remove those grave accents from the expression, the query would work.
Any suggestions on that?
